File: wpl/lab4/q2.py
#Write a python program to implement binary search with recursion.
def binary_search(arr, tar, low, high):
if low > high:
return -1
mid = (low + high) // 2
if arr[mid] == tar:Name : Princita Zina Miranda, Section : SCE-C, Reg no. 220905115 (roll no. 17)
return mid
if tar < arr[mid]:
return binary_search(arr, tar, low, mid - 1)
return binary_search(arr, tar, mid + 1, high)
def main():
arr = [12, 27, 38, 43, 76, 99]
tar = int(input("searching: "))
result = binary_search(arr, tar, 0, len(arr) - 1)
print("found at index:", result)
if __name__ == "__main__":
main()

File: VI Sem/cd-lab/lab2/q2.c
// Name : Princita Zina Miranda
// Section : SCE-C
// Reg No. : 220905115 (Roll No. 17)

// Write a C program to discard preprocessor directives from the given input 'C' file

#include <stdio.h>
#include <stdlib.h>
#define Pi 3

int main() {
	FILE *fi, *fo;
	int c;
	fi = fopen("q2.c", "r");
	if (fi == NULL) {
		printf("Cannot open file \n");
		exit(0); 
	}
	fo = fopen("q2out.c", "w");
	c = getc(fi);
	while (c != EOF) {
		if(c =='\"') {
			putc(c, fo);
			c = getc(fi);
			putc(c, fo);
			while(c != '\"') {
				c = getc(fi);
				putc(c, fo);
			}
		}
		else if(c == '#') {
			while(c != '\n')
				c = getc(fi);
		}
		else putc(c, fo);
		c = getc(fi);
	}
	fclose(fi);
	fclose(fo);
	return 0;
}
